geode-core/src/main/java/org/apache/geode/management/internal/cli/remote/CommandExecutor.java

```diff
@@ -131,16 +136,31 @@ protected Object invokeCommand(Object command, GfshParseResult parseResult) {
     }
 
     String groupInput = parseResult.getParamValueAsString("group");
+    String[] groups;
     if (groupInput == null) {
-      groupInput = "cluster";
+      groups = new String[] {"cluster"};
```

Comment:
I like the overall general approach, however I don't like that the code flow here does not make it clear that `updateCacheConfig` and `updateConfigForGroup` are mutually exclusive.

Reply:
Looks good with the new marker interface.